### ISSUE TITLE
seed-song option slowness / does not return

### ISSUE THREAD

**Issue by jcorporation** [Polochon-street/blissify-rs#63]
Hi, sorry that I must open another issue...

I just tested the --seed-song option as it sounds interesting. 

In my setup (current master) it does not return or is so slow that I can not await it. You mentioned in the README that this option is somewhat slower, but I think not so extremely slower?

```
time blissify playlist --dry-run --from-song "/musik/Alben/Smashing Pumpkins/Live at Township Auditorium on 2007-11-09/tsp2007-11-09t05.mp3" 10

real	 0m2,402s
user 0m2,335s
sys	 0m0,065s
```

With --seed-song option set. I aborted the task after 2 minutes of 100% CPU usage of blissify.
```
time blissify playlist --seed-song --dry-run --from-song "/musik/Alben/Smashing Pumpkins/Live at Township Auditorium on 2007-11-09/tsp2007-11-09t05.mp3" 10

real	 2m0,134s
user 2m0,173s
sys	 0m0,067s
```

I tested also without the --dry-run and --from-song options, but that does not help. Lowering the number of songs does also show no success.

**Comment by Polochon-street** [comment]
Hi,

Thanks for reporting. It's due to the fact that it computes a playlist with *all* the songs of the user's library, and then crops it to the desired number of songs.

I'm working on optimizing it, but it's not as trivial as it sounds, unfortunately. I'll keep you updated :)

**Comment by jcorporation** [comment]
Thanks, it would be a great feature!

**Comment by Polochon-street** [comment]
it took a fairly big rewrite of the playlist module, but it should be working faster now :)
I just published `blissify 0.4.1`, `--seed-song` should return faster there.

**Comment by jcorporation** [comment]
Great, I will test it soon.
